x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/WaitForActiveShardsStep.java

```diff
@@ -64,43 +65,48 @@ public Result isConditionMet(Index index, ClusterState clusterState) {
             return new Result(true, new Info(message));
         }
 
-        String rolloverAlias = RolloverAction.LIFECYCLE_ROLLOVER_ALIAS_SETTING.get(originalIndexMeta.getSettings());
-        if (Strings.isNullOrEmpty(rolloverAlias)) {
-            throw new IllegalStateException("setting [" + RolloverAction.LIFECYCLE_ROLLOVER_ALIAS
-                + "] is not set on index [" + originalIndexMeta.getIndex().getName() + "]");
-        }
-
-        IndexAbstraction indexAbstraction = clusterState.metadata().getIndicesLookup().get(rolloverAlias);
-        assert indexAbstraction.getType() == IndexAbstraction.Type.ALIAS : rolloverAlias + " must be an alias but it is not";
-
-        IndexMetadata aliasWriteIndex = indexAbstraction.getWriteIndex();
+        IndexAbstraction indexAbstraction = clusterState.metadata().getIndicesLookup().get(index.getName());
         final String rolledIndexName;
         final String waitForActiveShardsSettingValue;
-        if (aliasWriteIndex != null) {
-            rolledIndexName = aliasWriteIndex.getIndex().getName();
-            waitForActiveShardsSettingValue = aliasWriteIndex.getSettings().get("index.write.wait_for_active_shards");
+        if (indexAbstraction.getParentDataStream() != null) {
+            DataStream dataStream = indexAbstraction.getParentDataStream().getDataStream();
+            rolledIndexName = DataStream.getBackingIndexName(dataStream.getName(), dataStream.getGeneration());
```

Comment:
Minor: this correctly identifies the data stream's write index under the current implementation of data streams. The `IndexAbstraction::getWriteIndex` method will always return the correct write index in the event that we change the logic around backing index names, generations, etc.

Reply:
Thanks for the suggestion, Dan. Pushed a fix to change this.